### PR TITLE
Make TestParticleMakie as an extension

### PR DESCRIPTION
**Pull request by TCLiuu**
1. Making TestParticleMakie as an extension #88 
2. Fix a bug #81

### CODE REVIEW

**Comment by TCLiuu** [comment]
Should we remove `TestParticleMakie` from registry?

**Comment by TCLiuu** [comment]
There is still a issue. Should we merge the testCI of TestParticlemakie and TestParticle? If not, the coverage report will have some strange problems when the testCI of TestParticlemakie does not run, as it occurred in those PRs before.

**Comment by henry2004y** [comment]
> Should we remove `TestParticleMakie` from registry?

If I remember correctly, there is currently no way of removing a package from the registry. We can simply mark it as archived and leave it there.


> There is still an issue. Should we merge the testCI of TestParticlemakie and TestParticle? If not, the coverage report will have some strange problems when the testCI of TestParticlemakie does not run, as it occurred in those PRs before.

I think we should merge the visual tests into the main tests. With CairoMakie and the CI cache system, this should not add too much test time. See an example [here](https://github.com/JuliaGeometry/Meshes.jl/blob/master/test/runtests.jl). They list CairoMakie as a test Project.toml dependency and always import it.

File: docs/src/plotfunctions.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Plot Functions
 
-Before using the plot recipes of `Testparticle.jl`, you need to install the package [`TestParticleMakie.jl`](https://github.com/henry2004y/TestParticle.jl/tree/master/TestParticleMakie) and import it via `using TestParticleMakie`. All plot recipes depend on `Makie.jl`.
+Before using the plot recipes of `Testparticle.jl`, you need to import `Makie` package via `using Makie`. All plot recipes depend on `Makie.jl`.
```
Comment (henry2004y) [inline docs/src/plotfunctions.md:3]: I think `using GLMakie` or `using CairoMakie` also implicitly import `Makie` itself, so we don't necessarily need to `using Makie`.

File: ext/typerecipes.jl [hunk 2]
```diff
@@ -46,6 +46,9 @@ function Makie.convert_arguments(P::PointBased, sol::AbstractODESolution;
     t_step = (t_max - t_min) / len
     t = collect(t_min:t_step:t_max)
     u = sol.(t)
+    if !isa(u[1], Vector)
```
Comment (henry2004y) [inline ext/typerecipes.jl:49]: I guess this fixed #81? Better to add a comment here.

**Comment by henry2004y** [comment]
This is almost done! Nice job!

**Review by henry2004y — APPROVED** [review]

**Pull request merged**